topo/node/drivenets/drivenets_test.go

```diff
@@ -0,0 +1,117 @@
+package drivenets
+
+import (
+	"testing"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/kne/topo/node"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name      string
+		nodeImpl  *node.Impl
+		wantErr   bool
+		errString string
+	}{
+		{
+			name:      "nil nodeImpl",
+			nodeImpl:  nil,
+			wantErr:   true,
+			errString: "nodeImpl cannot be nil",
+		},
+		{
+			name: "nil Proto",
+			nodeImpl: &node.Impl{
+				Proto: nil,
+			},
+			wantErr:   true,
+			errString: "nodeImpl.Proto cannot be nil",
+		},
+		{
+			name: "no model specified",
+			nodeImpl: &node.Impl{
+				Proto: &tpb.Node{},
+			},
+			wantErr:   true,
+			errString: "unknown model",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := New(tt.nodeImpl)
+			if tt.wantErr {
```

Comment:
this could be streamlined quite a bit using the errdiff package, see other examples in this repo: https://github.com/openconfig/kne/blob/f197e16f84e47b8eadc4263d71fafe92b28dfbd9/exec/run/run_test.go#L108

Reply:
i modifed to use `errdif`